pandas/io/common.py

```diff
@@ -85,7 +88,7 @@ def _expand_user(
     expanded_filepath_or_buffer : an expanded filepath or the
                                   input if not expandable
     """
-    if isinstance(filepath_or_buffer, str):
+    if isinstance(filepath_or_buffer, (str, pathlib.Path)):
```

Comment:
is this a change (positive one)?

Reply:
I think this is never called with Paths so it doesn't make a practical difference. I did it to narrow down the possible return types.

Reply:
I guess that since `FileOrBuffer` alias has been added in this PR, the input to _expand_user could be refined instead. either way works for me.

if adding pathlib.Path here,we may want to remove 

```
    elif isinstance(filepath_or_buffer, pathlib.Path):	
        return str(filepath_or_buffer)
```
from stringify_path (but I think the early returns there maybe a bug since Path objects aren't expanded)

see also https://github.com/pandas-dev/pandas/pull/35401 for work in this area (we currently support the \_\_fspath__ protocol directly, that PR aims to use the library to support \_\_fspath__ instead)

Reply:
thanks @simonjayhawkins I will look into that! I just noticed the (non-)expand inconsistency in `stringify_path` when I extended a test to cover str, Path, and a `fspath` object. I removed the early return statements from `_expand_user`.

Reply:
about `FileOrBuffer` I'm fine with removing it. I added it as the returned `filepath_or_buffer` of `get_filepath_or_buffer` can never be a `Path`.

Reply:
no I didn't mean remove it.

I meant the signature could be `def _expand_user(filepath_or_buffer: FileOrBuffer[AnyStr]) -> FileOrBuffer[AnyStr]:` instead.

since `_expand_user` is only called from `stringify_path` and hence a Path object is never passed

Reply:
okay, yes that makes sense!

Reply:
`_expand_user` is called from pandas/io/parquet.py as well which might call it with a Path :(

Reply:
hmm, that's been added recently. I wonder whether Path objects and objects with the \_\_fspath__ protocol are properly supported there. The call should maybe have been to `stringify_path`. `_expand_user` is private after all.

Reply:
I changed `_expand_user` to `stringify_path`